relayer/naive-strategy.go

```diff
@@ -232,6 +238,36 @@ func (nrs *NaiveStrategy) UnrelayedAcknowledgements(src, dst *Chain) (*RelaySequ
 
 // HandleEvents defines how the relayer will handle block and transaction events as they are emitted
 func (nrs *NaiveStrategy) HandleEvents(src, dst *Chain, events map[string][]string) {
+	// check for misbehaviour
+	if hdrs, ok := events[fmt.Sprintf("%s.%s", updateCliTag, headerTag)]; ok {
+		for i, hdr := range hdrs {
+			clientIDs := events[fmt.Sprintf("%s.%s", updateCliTag, clientIDTag)]
+
+			if src.PathEnd.ClientID == clientIDs[i] && dst.PathEnd.ClientID == clientIDs[i] {
+				hdrBytes, err := hex.DecodeString(hdr)
+				if err == nil {
+					exportedHeader, err := clienttypes.UnmarshalHeader(src.Encoding.Marshaler, hdrBytes)
+					if err == nil {
+						gotHeader, ok := exportedHeader.(*tmclient.Header)
+						if ok {
+							trustedHeader, err := src.GetLatestLightHeader()
+							if err == nil && !reflect.DeepEqual(gotHeader, trustedHeader) {
+								misbehaviour := tmclient.NewMisbehaviour(clientIDs[i], gotHeader, trustedHeader)
```

Comment:
lets make `clientIDs[i]` a var at the beginning of the for, maybe a name like `emittedClientID`